sormas-ui/src/main/java/de/symeda/sormas/ui/sormastosormas/ShareRequestGridComponent.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2021 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.symeda.sormas.ui.sormastosormas;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import com.vaadin.navigator.ViewChangeListener;
+import com.vaadin.ui.Button;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.ui.themes.ValoTheme;
+
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.sormastosormas.sharerequest.ShareRequestCriteria;
+import de.symeda.sormas.api.sormastosormas.sharerequest.ShareRequestStatus;
+import de.symeda.sormas.ui.utils.ButtonHelper;
+import de.symeda.sormas.ui.utils.CssStyles;
+import de.symeda.sormas.ui.utils.LayoutUtil;
+import de.symeda.sormas.ui.utils.ViewConfiguration;
+
+public class ShareRequestGridComponent extends VerticalLayout {
+
+	private static final long serialVersionUID = 99959936905302170L;
+	private ShareRequestCriteria criteria;
+	private ShareRequestFilterForm filterForm;
+	private ShareRequestGrid grid;
+	private Map<Button, String> statusButtons;
+	private Button activeStatusButton;
+
+	public ShareRequestGridComponent(
+		ViewConfiguration viewConfiguration,
+		ShareRequestCriteria criteria,
+		Runnable filterChangeHandler,
+		Runnable filterResetHandler) {
+		this.criteria = criteria;
+
+		setSizeFull();
+		setMargin(false);
+
+		grid = new ShareRequestGrid(viewConfiguration.isInEagerMode(), criteria);
+		grid.setSizeFull();
+		grid.getDataProvider().addDataProviderListener(e -> updateStatusButtons());
+
+		VerticalLayout gridLayout = new VerticalLayout();
+		/* gridLayout.addComponent( */createFilterBar(filterChangeHandler, filterResetHandler)/* ) */;
```

Comment:
Maybe also add a comment here that the filter bar will only be added once more filters are introduced.